src/resolved.ts

```diff
@@ -48,12 +72,12 @@ export class Resolved {
 
     if (target && target[REF_METADATA]) {
       return {
-        path: [...get(target, [REF_METADATA, 'root'], []), ...newPath],
+        path: [...get(target, [REF_METADATA, 'root'], []).map(decodePointerFragment), ...newPath],
```

Comment:
Why is `decodePointerFragment` required now?

Reply:
So the path stored under `get(target, [REF_METADATA, 'root']` is escaped, meaning `/foo` is represented as `~1foo`.
The path we expect to return should be unescaped (`/foo`), so that the YAML/JSON getLocationForJsonPath can find a node correctly.
If you revert the change, the test I added will fail - produced ranges will be less precise (they will point at `paths` rather than `paths//test/get`.